File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> common) {
    int last = common[common.size() - 1];
    int dif = common[1] - common[0];
    int mul;
    
    if(common[0] == 0 && dif == 0)
        return 0;
    
    mul = common[2] / common[1];
    
    for(int i = 1; i < common.size(); i++)
    {
        if(common[i] - common[i-1] != dif)
            return last * mul;
    }
    
    return last + dif;
}

File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.cs
using System;

public class Solution {
    public int solution(int[] common) {
        int answer = 0;
        bool isAdd = true;
        int mul = 1;
        int dif = common[1] - common[0];
        int last = common[common.Length - 1];
        for(int i = 1; i < common.Length; i++)
        {
            if(common[i] != 0 && common[i-1] != 0)
                mul = common[i] / common[i-1];
            
            if(common[i] - common[i-1] == dif)
                isAdd = true;
            else
            {
                isAdd = false;
                break;
            }
        }
        answer = isAdd == true ? last + dif : last * mul;
        return answer;
    }
}